### PR TITLE
Add creation of a necessary GCP firewall rule to clusterctl/example/google/generate-yaml.sh

### PR DESCRIPTION
**Pull request by spew**
**What this PR does / why we need it**:
This PR makes it easier to get up and running by creating the necessary firewall rule for you. It is a temporary stopgap until we do this automatically.

**Release note**:
```release-note
NONE
```

@kubernetes/kube-deploy-reviewers

### CODE REVIEW

File: clusterctl/examples/google/generate-yaml.sh [hunk 1]
```diff
@@ -119,6 +119,12 @@ gcloud projects add-iam-policy-binding $GCLOUD_PROJECT --member=serviceAccount:$
 echo Generating $WORKER_SA_EMAIL service account for workers
 gcloud iam service-accounts create --display-name="worker service account" $WORKER_SA_NAME
 
+FIREWALL_RULE_NAME="cluster-api-open"
+if ! gcloud compute firewall-rules describe ${FIREWALL_RULE_NAME}; then
+  echo "Creating ${FIREWALL_RULE_NAME} firewall rule to enable inbound communication to nodes"
+  gcloud compute firewall-rules create ${FIREWALL_RULE_NAME} --allow=TCP:443 --source-ranges=0.0.0.0/0 --target-tags='https-server'
```
Comment (mkjelland) [inline clusterctl/examples/google/generate-yaml.sh]: Do we also need to allow traffic between the nodes?

**Comment by mkjelland** [comment]
lgtm, leaving for @jessicaochen to add the label

File: clusterctl/examples/google/generate-yaml.sh [hunk 2]
```diff
@@ -119,6 +119,12 @@ gcloud projects add-iam-policy-binding $GCLOUD_PROJECT --member=serviceAccount:$
 echo Generating $WORKER_SA_EMAIL service account for workers
 gcloud iam service-accounts create --display-name="worker service account" $WORKER_SA_NAME
 
+FIREWALL_RULE_NAME="cluster-api-open"
```
Comment (jessicaochen) [inline clusterctl/examples/google/generate-yaml.sh]: (nit)
Perhaps keep this const with the others at the top of the file?

File: clusterctl/examples/google/generate-yaml.sh [hunk 3]
```diff
@@ -119,6 +119,12 @@ gcloud projects add-iam-policy-binding $GCLOUD_PROJECT --member=serviceAccount:$
...
+if ! gcloud compute firewall-rules describe ${FIREWALL_RULE_NAME}; then
```
Comment (jessicaochen) [inline clusterctl/examples/google/generate-yaml.sh]: (nit)
All the other conditions seem to be using brackets. Can we use it here too for consistency?

**Pull request merged**